test/0.8.9/withdrawal-queue-requests-finalization.test.js

```diff
@@ -71,6 +71,75 @@ contract('WithdrawalQueue', ([owner, daoAgent, user, anotherUser]) => {
     await snapshot.rollback()
   })
 
+  context('calculateFinalizationBatches', () => {
+    afterEach(async () => {
+      await snapshot.rollback()
+    })
+    it('reverts on invalid state', async () => {
+      await assert.reverts(
+        withdrawalQueue.calculateFinalizationBatches(shareRate(300), 100000, 1000, [
+          ETH(10),
+          true,
+          Array(36).fill(0),
+          0,
+        ]),
+        'InvalidState()'
+      )
+      await assert.reverts(
+        withdrawalQueue.calculateFinalizationBatches(shareRate(300), 100000, 1000, [0, false, Array(36).fill(0), 0]),
```

Comment:
can we possibly leverage dedicated types (structs) here and there for such long args lists?